core/src/worker/workflow/machines/workflow_machines.rs

```diff
@@ -318,6 +318,17 @@ impl WorkflowMachines {
         if event.is_final_wf_execution_event() {
             self.have_seen_terminal_event = true;
         }
+        if event.event_type() == EventType::WorkflowExecutionTerminated {
```

Comment:
Would there be other possible events here?

Reply:
No - the other terminal events are already handled via the normal mechanisms - since completing the workflow any other way is like issuing a command and has the normal determinism constraints.

Reply:
Although there is timed out which isn't like that. Thanks for mentioning